src/containers/Header/Search.jsx

```diff
@@ -17,16 +17,30 @@ import {
   HASH_REGEX,
 } from '../shared/utils'
 import './search.scss'
+import { getNFTInfo } from '../../rippled/lib/rippled'
+import SocketContext from '../shared/SocketContext'
 
-const getIdType = (id) => {
+const isNFTOrTransactions = async (id, rippledContext) => {
```

Comment:
Is there not a difference in the length/structure that can be used to identify whether something is an NFT or a transaction?

Reply:
Not that Im aware of, both seem to follow the same length. This was the approach discussed with @ledhed2222. But I would be happy to know if there exists any hacks around this

Reply:
NFTs do have to start with `000`, but a transaction hash could potentially also start with `000` (though it would be pretty rare - 1/4096 chance). So I guess this is the best option.

Reply:
Why do NFTs always start with 000?

Reply:
Im not sure if it sounds like a good approach tho. still could collide...

Reply:
If you look at how an NFT ID is derived, it starts with the the flags (the first 4 characters, in hex). There are no flags on an NFT with value greater than 8, so therefore the first 6 bytes (and the first 3 characters) are `0`.

Reply:
Although this won't work as a general case, we could save some time by checking the first 3 characters - if it doesn't start with `000`, then we know it's a transaction.

Reply:
👍

Reply:
@ckniffen @mvadari i see there is a reserved flag for the first 3 bits that could change in the future, so im skeptical about checking the first 3 bits

Reply:
We could check the first 3 bits now , and remember to change it if the reserved flag is used